src/api/test/functional/maintenance_test.rb

```diff
@@ -1473,9 +1473,10 @@ def test_create_maintenance_project_and_release_packages
     run_publisher
     # published binaries from incident got removed?
     get "/published/#{incident_project}/BaseDistro3/i586/package-1.0-1.i586.rpm"
-    assert_response 404
+    assert_response 200
```

Comment:
How can this now be 200 if there is no delay in removing (as before)?

Reply:
As soon as I move the cleanup to a job, the files are found (so 200 is returned), no matter if the job is performed with or without delay. Even if the queue adapter is `:inline`.

Apart from that, my plan was to emulate the case as similar to production as possible. For that, if I set a delay like in `PublishedRepositoriesCleanupJob.set(wait: 3.seconds).perform_later(project_name)`, I need to use `:async` as queue adapter. Otherwise, I'll get an error like:

```
NotImplementedError (Use a queueing backend to enqueue jobs in the future.
```

https://api.rubyonrails.org/classes/ActiveJob/QueueAdapters/InlineAdapter.html
https://api.rubyonrails.org/classes/ActiveJob/QueueAdapters/AsyncAdapter.html

Reply:
> As soon as I move the cleanup to a job, the test finds the files

Why?

Reply:
Well, I've finally decided not to change the queue adapter, so it will keep using `:inline` since it's the one in the setup.
In fact, now that the option `# maintenance_release_repositories_lifetime: 172800` is commented/optional, it won't be used in CircleCI tests, so I emulate the same locally.

Regarding why the cleanup job find the files if there isn't any waiting time and `:inline`is set up... I'm still trying to figure out.